Horizon/Horizon/Sources/Features/Notebook/Common/Data/CourseNotesRepository.swift

```diff
@@ -20,184 +20,283 @@ import Combine
 import Foundation
 
 protocol CourseNotesRepositoryProtocol {
+    func delete(id: String) -> Future<Void, Error>
     func get() -> AnyPublisher<[CourseNote], any Error>
+    func get(id: String) -> AnyPublisher<CourseNote?, any Error>
+    func set(id: String, content: String?, labels: [CourseNoteLabel]?) -> Future<Void, Error>
 }
 
-struct CourseNotesRepository: CourseNotesRepositoryProtocol {
+struct RepositoryNote {
+    let id: String
+    let date: Date
+    let content: String
+    let institution: String
+    let courseId: String
+    let course: String
+    let labels: [String]
+}
+
+extension RepositoryNote {
+    func toCourseNote() -> CourseNote {
+        CourseNote(
+            id: self.id,
+            date: self.date,
+            content: self.content,
+            institution: self.institution,
+            courseId: self.courseId,
+            course: self.course,
+            labels: self.labels.map { CourseNoteLabel(rawValue: $0) ?? .other }
+       )
+    }
+}
+
+class StaticCourseNotesRepository: CourseNotesRepositoryProtocol {
+    // MARK: - Static
```

Comment:
I'm confuse about you name the repo `static` , this refer to dummy data for what ?

Reply:
Where this came from is I needed a single instance of the CourseNotesRepository across multiple pages so I created a singleton. The label `static` just refers to the statically defined property.